src/components/panel/panel.js

```diff
@@ -682,52 +709,80 @@ function MdPanelService($rootElement, $rootScope, $injector, $window) {
   /** @private @const */
   this._$window = $window;
 
+  /** @private {!Object<string, !MdPanelRef>} */
+  this._trackedPanels = {};
 
   /**
+   * @description
    * Default animations that can be used within the panel.
+   *
    * @type {enum}
    */
   this.animation = MdPanelAnimation.animation;
 
   /**
+   * @description
    * Possible values of xPosition for positioning the panel relative to
    * another element.
+   *
    * @type {enum}
    */
   this.xPosition = MdPanelPosition.xPosition;
 
   /**
+   * @description
    * Possible values of yPosition for positioning the panel relative to
    * another element.
+   *
    * @type {enum}
    */
   this.yPosition = MdPanelPosition.yPosition;
 }
 
 
 /**
+ * @description
  * Creates a panel with the specified options.
+ *
  * @param {!Object=} config Configuration object for the panel.
  * @returns {!MdPanelRef}
  */
 MdPanelService.prototype.create = function(config) {
   var configSettings = config || {};
 
+  if (angular.isDefined(configSettings.id) &&
+      angular.isDefined(this._trackedPanels[configSettings.id])) {
+        return this._trackedPanels[configSettings.id];
+      }
+
   this._config = {
     scope: this._$rootScope.$new(true),
     attachTo: this._$rootElement
   };
   angular.extend(this._config, this._defaultConfigOptions, configSettings);
 
-  var instanceId = 'panel_' + this._$injector.get('$mdUtil').nextUid();
-  var instanceConfig = angular.extend({ id: instanceId }, this._config);
+  var instanceConfig;
+  if (angular.isUndefined(this._config.id)) {
+    var instanceId = 'panel_' + this._$injector.get('$mdUtil').nextUid();
```

Comment:
`$mdUtil` is injected directly into the MdPanelRef service as well as the MdPanelAnimation service. This is the only place that it is used inside of the MdPanelService API.